### PR TITLE
docs(graphql): update GraphQL docs

### LINKED ISSUE #878: [Task] Documentation for GraphQL
GraphQL for 2.1.0 is feature-complete after #872 is merged. We need upstream development-level docs for this feature, including an entry in `HTTP_API.md`.

https://github.com/cryostatio/cryostat/pull/872#issuecomment-1088970909

It probably makes more sense to document the various queries/mutations implemented and the possible filter inputs and the behaviour of using multiple filter input fields somewhere separately from `HTTP_API.md`, since that's a lot of content specific to just one high-powered API endpoint. The existing `GRAPHQL.md` could be a good candidate for reuse here.

### PR DESCRIPTION
**Pull request by andrewazores**
- docs(graphql): move graphql examples into docs/
- allow user to use devmode and non-basic auth
- update graphql docs

Fixes #878

### CODE REVIEW

File: docs/GRAPHQL.md [hunk 1]
```diff
@@ -5,20 +5,23 @@
 ### `[GET|POST] /api/v2.2/graphql`
 
 Accepts `GET` or `POST` requests to perform GraphQL queries. See:
-- https://www.graphql-java.com/tutorials/getting-started-with-spring-boot/
-- https://graphql.org/learn/serving-over-http/
+- [https://www.graphql-java.com/tutorials/getting-started-with-spring-boot/](Getting Started)
```
Comment (tthvo) [inline docs/GRAPHQL.md:8]: I guess it has to be in `[text](link)` format?

Screenshot from 2022-09-20 11-21-22

File: docs/GRAPHQL.md [hunk 2]
```diff
@@ -5,20 +5,23 @@
...
-example queries. The first query is a standard API v2 requst to create a custom
+[`graphql/`](graphql/) contains some sample queries which can be used as in the following
+example queries. The first query is a standard API v2 request to create a custom
 target. The second query is a standard API v1 request to list all known targets.
 The third query is a GraphQL query listing all known targets and all fields of
 those targets, except without querying for the recordings belonging to the
 target.
 
+For more information about the available queries and mutations check the
+GraphiQL interface detailed below, or check the GraphQL schema definitions:
```
Comment (tthvo) [inline docs/GRAPHQL.md:20]: typo at `GraphiQL`?

Comment (tthvo) [reply docs/GRAPHQL.md:20]: Probably we should add a comma at `...mutations, check...`

File: docs/GRAPHQL.md [hunk 3]
```diff
@@ -131,137 +134,6 @@ Serves a GraphQL "query IDE" that can be used for testing out writing queries
...
-### Quick Reference
-
-| What you want to do                                                       | Which handler you should use                                                    |
-| ------------------------------------------------------------------------- | --------------------------------------------------------------------------------|
-| **Recordings in Target JVMs**                                             |                                                                                 |
-| Update metadata for a recording in a target JVM                    | [`PutActiveRecordingMetadataMutator`](#PutActiveRecordingMetadataMutator) |
-| **Recordings in archive**                                                 |                                                                                 |
-| Update metadata for an archived recording                          | [`PutArchivedRecordingMetadataMutator`](#PutArchivedRecordingMetadataMutator)     |
-
-### Recordings in Target JVMs
```
Comment (maxcao13) [inline docs/GRAPHQL.md:147]: Should these types of graphql endpoint querying docs be put somewhere else or are we just removing them altogether?

File: run.sh [hunk 4]
```diff
@@ -138,4 +138,5 @@ podman run \
     -e CRYOSTAT_HIBERNATE_DIALECT="$CRYOSTAT_HIBERNATE_DIALECT" \
     -e CRYOSTAT_HBM2DDL="$CRYOSTAT_HBM2DDL" \
     -e CRYOSTAT_LOG_DB_QUERIES="true" \
+    -e CRYOSTAT_DEV_MODE="$CRYOSTAT_DEV_MODE" \
```
Comment (maxcao13) [inline run.sh:141]: What does the dev mode do and how do we use it?

File: docs/GRAPHQL.md (hunk 1 above)
Comment (andrewazores) [reply docs/GRAPHQL.md:8]: Haha, whoops. Thanks.

File: docs/GRAPHQL.md (hunk 2 above)
Comment (andrewazores) [reply docs/GRAPHQL.md:20]: No, [`GraphiQL`](https://github.com/graphql/graphiql) is intentional. It's the name of the "IDE" view.

File: docs/GRAPHQL.md (hunk 3 above)
Comment (andrewazores) [reply docs/GRAPHQL.md:147]: I'm removing them altogether. I don't think it's worth maintaining them. End users who want to interact with the GraphQL API should either use the GraphiQL "IDE" or read the schemas. Hand-written docs like our HTTP_API.md are prone to falling out of date, but the schema is always a correct source of information.

File: run.sh (hunk 4 above)
Comment (andrewazores) [reply run.sh:141]: `CRYOSTAT_DEV_MODE=true CRYOSTAT_AUTH_MANAGER=io.cryostat.net.NoopAuthManager sh smoketest.sh`, then in your browser open `https://localhost:8181/api/v2.2/graphiql/`.

`sh devserver.sh` also sets the `CRYOSTAT_DEV_MODE=true`.

Currently the GraphiQL endpoint is the only thing gated behind the dev mode variable.

**Pull request merged**